apollo-server/v3/integration-testing/src/server.ts

```diff
@@ -0,0 +1,34 @@
+import { ApolloServer, gql } from 'apollo-server';
+
+// Schema definition
+export const typeDefs = gql`
+  type Query {
+    hello(name: String): String!
+  }
+`;
+
+// Resolver map
+export const resolvers = {
+  Query: {
+    hello: (_, { name }) => `Hello ${name}!`,
+  },
+};
+
+export const createApolloServer = async (options = { port: 4000 }) => {
+  const server = new ApolloServer({
+    typeDefs,
+    resolvers,
+  });
+
+  return server
+    .listen(options, () => {
+      if (process.env.NODE_ENV !== 'test') {
+        console.log(
```

Comment:
Because we've made *our* `listen` function into an async function, we can do this logging after just awaiting server.listen instead of via this callback that I didn't even realize would work.

So:

```
const serverInfo = await server.listen(options);
if (not test) {console.log()}
return serverInfo;
```

Reply:
Oh, that's neat! I really like how clear that looks too.